Framework/DataObjects/inc/MantidDataObjects/EventWorkspace.h

```diff
@@ -127,6 +127,16 @@ class DLLExport EventWorkspace : public API::IEventWorkspace {
   Kernel::cow_ptr<HistogramData::HistogramX>
   refX(const std::size_t) const override;
 
+  HistogramData::HistogramY &mutableY(const size_t index) override {
+    throw Mantid::Kernel::Exception::NotImplementedError(
+        "This method has not been implemented for EventWorkspace");
+  }
+
+  HistogramData::HistogramE &mutableE(const size_t index) override {
+    throw Mantid::Kernel::Exception::NotImplementedError(
+        "This method has not been implemented for EventWorkspace");
+  }
+
```

Comment:
@LamarMoore Probably we can make all methods on `MatrixWorkspace` non-virtual (those of the `Histogram` interface). The all just call `getSpectrum()`, and we can then handle things via virtual methods on `ISpectrum`.